### PR TITLE
mc watch windows URL/Absolute path event path 

### PR DESCRIPTION
**Pull request by BigUstad**
Folder is an URI/URL but rename/write event path is absolute path

### CODE REVIEW

File: cmd/mirror-main.go [hunk 1]
```diff
@@ -405,7 +405,27 @@ func (mj *mirrorJob) watchMirror(ctx context.Context, cancelMirror context.Cance
...
+				shdModifyEventPath := filepath.IsAbs(eventPath) && !filepath.IsAbs(mj.sourceURL) && lenSrcURLSlice > 1
+				if shdModifyEventPath {
+					lastPathPrefix = sourceURLpathList[lenSrcURLSlice-1]
+					lastPathPrefixPos = strings.Index(eventPath, lastPathPrefix)
+				}
+				canModifyEventPath := shdModifyEventPath && lastPathPrefix != "" && lastPathPrefixPos > 0
+				canModifyEventPath = canModifyEventPath && lastPathPrefixPos+len(lastPathPrefix) < len(eventPath)
+				if canModifyEventPath {
+					eventRelPath = eventPath[lastPathPrefixPos+len(lastPathPrefix):]
+					eventRelPath = strings.Replace(eventRelPath, "\\", "/", -1)
```
Comment (harshavardhana) [inline cmd/mirror-main.go]: filepath.ToSlash() is the helper you want @BigUstad

File: cmd/client-fs.go [hunk 2]
```diff
@@ -183,6 +183,14 @@ func (f *fsClient) Watch(params watchParams) (*watchObject, *probe.Error) {
 				i, e = os.Stat(event.Path())
 				if e != nil {
 					if os.IsNotExist(e) {
+						if runtime.GOOS == "windows" && event.Event()&notify.Rename != 0 {
```
Comment (harshavardhana) [inline cmd/client-fs.go]: Does this perhaps need to be done outside? because looking for delete event can be done before doing a stat on the path?

Comment (BigUstad) [reply cmd/client-fs.go]: This is for the case of renaming of a file (notify.Rename). The event (notify.Rename) is generated with the older filename and the newer(changed) filename. We need this check to distinguish between the older & newer names. Please advise.

File: cmd/mirror-main.go (hunk 1 above)
Comment (BigUstad) [reply cmd/mirror-main.go]: Thanks.

File: cmd/client-fs.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/client-fs.go]: Correct but it's added deep down after os.Stat() why not before?

File: cmd/client-fs_windows.go [hunk 3]
```diff
@@ -37,7 +37,9 @@ func IsGetEvent(event notify.Event) bool {
 // IsPutEvent checks if the event returned is a put event
 func IsPutEvent(event notify.Event) bool {
 	for _, ev := range EventTypePut {
-		if event&ev != 0 {
+		if event&notify.FileActionRenamedOldName != 0 {
```
Comment (kannappanr) [inline cmd/client-fs_windows.go]: This has to be outside the `for` loop, why would you check this every time

**Comment by BigUstad** [comment]
There was a mistake in the git push. Closing this PR. Will open another PR.

**Pull request closed without merging**